test/unittest/tests/block/test_block_api_misc.cpp

```diff
@@ -0,0 +1,257 @@
+#include <string>
+
+#include "wt_internal.h"
+#include "util_block.h"
+#include "../utils_extlist.h"
+#include "../wrappers/config_parser.h"
+#include "../wrappers/mock_session.h"
+
+const std::string ALLOCATION_SIZE = "256";
+const std::string BLOCK_ALLOCATION = "best";
+const std::string OS_CACHE_MAX = "0";
+const std::string OS_CACHE_DIRTY_MAX = "0";
+const std::string ACCESS_PATTERN = "random";
+const std::string DEFAULT_FILE_NAME = "test.txt";
+
+static void
+check_bm_stats(WT_SESSION_IMPL *session, WT_BM *bm)
+{
+    WT_DSRC_STATS stats;
+
+    // Enable statistics on the connection to allow the statistic macros to update correctly.
+    S2C(session)->stat_flags = 1;
+    REQUIRE(bm->stat(bm, session, &stats) == 0);
+    CHECK(stats.allocation_size == bm->block->allocsize);
+    CHECK(stats.block_checkpoint_size == (int64_t)bm->block->live.ckpt_size);
+    CHECK(stats.block_magic == WT_BLOCK_MAGIC);
+    CHECK(stats.block_major == WT_BLOCK_MAJOR_VERSION);
+    CHECK(stats.block_minor == WT_BLOCK_MINOR_VERSION);
+    CHECK(stats.block_reuse_bytes == (int64_t)bm->block->live.avail.bytes);
+    CHECK(stats.block_size == bm->block->size);
+
+    // Disable statistics on the connection when finished so that the mock session destructor
+    // doesn't try to dereference invalid memory.
+    S2C(session)->stat_flags = 0;
+}
+
+static int
+test_addr_invalid(WT_SESSION_IMPL *session, WT_BM *bm, wt_off_t pack_offset, uint32_t pack_size,
+  uint32_t pack_checksum)
+{
+    // Generate an address cookie - technically, we shouldn't know about internal details of the
+    // address cookie, but this allows for more rigorous testing with different inputs.
+    uint8_t p[WT_BTREE_MAX_ADDR_COOKIE], *pp;
+    pp = p;
+    REQUIRE(__wt_block_addr_pack(
+              bm->block, &pp, WT_TIERED_OBJECTID_NONE, pack_offset, pack_size, pack_checksum) == 0);
+    size_t addr_size = WT_PTRDIFF(pp, p);
+
+    return (bm->addr_invalid(bm, session, p, addr_size));
+}
+
+// Test that the block manager's addr_string method produces the expected string representation.
+static void
+test_addr_string(WT_SESSION_IMPL *session, WT_BM *bm, wt_off_t pack_offset, uint32_t pack_size,
+  uint32_t pack_checksum, std::string expected_str)
+{
+    // Initialize a buffer.
+    WT_ITEM buf;
+    WT_CLEAR(buf);
+
+    // Generate an address cookie - technically, we shouldn't know about internal details of the
+    // address cookie, but this allows for more rigorous testing with different inputs.
+    uint8_t p[WT_BTREE_MAX_ADDR_COOKIE], *pp;
+    pp = p;
+    REQUIRE(__wt_block_addr_pack(
+              bm->block, &pp, WT_TIERED_OBJECTID_NONE, pack_offset, pack_size, pack_checksum) == 0);
+    size_t addr_size = WT_PTRDIFF(pp, p);
+
+    // Compare the string output of bm->addr_string against the known expected string.
+    REQUIRE(bm->addr_string(bm, session, &buf, p, addr_size) == 0);
+    CHECK(static_cast<std::string>(((char *)(buf.data))).compare(expected_str) == 0);
```

Comment:
Too many levels of ().  static_cast<std::string>((char *) buf.data) works.

Reply:
To use strcmp it would have to be through here:
`CHECK(strcmp(reinterpret_cast<char *>(const_cast<void*>(buf.data)), expected_str.c_str()) == 0);`

I don't think its much cleaner. In terms of efficiency, I do believe it is better, but not sure if it's worth. What do you think @tod-johnson-mongodb